52.md

```diff
@@ -0,0 +1,130 @@
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant
+* `t` (optional, repeated) hashtag to categorize calendar event
+* `r` (optional, repeated) references / links to web pages, documents, video calls, recorded videos, etc.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": "31923",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["summary", "<summary of calendar event>"],
+    ["description", "<description of calendar event>"],
+
+    // Timestamps
+    ["start", "<unix timestamp in seconds>"],
+    ["end", "<unix timestamp in seconds>"],
+
+    // Location
+    ["location", "<location>"],
+    ["g", "<geohash>"],
+
+    // Participants
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+
+    // Hashtags
+    ["t", "<tag>"],
+    ["t", "<tag>"],
+
+    // Reference links
+    ["r", "<url>"],
+    ["r", "<url>"]
+  ]
+}
+```
+
+Calendar events are not required to be part of a [calendar](#calendar).
+
+## Calendar Event RSVP
+
+A calendar event RSVP is a response to a calendar event to indicate a user's attendance intention.
+
+If a calendar event tags a pubkey, that can be interpreted as the calendar event creator is inviting that user to be an attendee. Clients MAY choose to prompt the user to RSVP for the calendar event.
+
+Any user may RSVP, even if they were not tagged on the calendar event. Clients MAY choose to prompt the calendar event creator to invite the user who RSVP'd. Clients also MAY choose to ignore these RSVPs.
+
+This NIP is intentionally not defining who is authorized to attend a calendar event if the user who RSVP'd has not been tagged. It is up to the calendar event creator to determine the semantics.
+
+### Format
+
+The format uses a parameterized replaceable event kind `31924` with a list of tags as described below:
+* `a` (required) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+* `d` (required) universally unique identifier. Generated by the client creating the calendar event RSVP.
+* `t` (required) attendance status. It can be one of the following values: `yes`, `no`, `maybe`
+* `t` (optional) availability status. It can be one of the following values: `free`, `busy`
+* `note` (optional) free-form note that adds more context to this calendar event response.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": "31924",
+  "tags": [
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["d", "<UUID>"]
+
+    ["t", "<yes/no/maybe>"],
```

Comment:
I was thinking that some clients may find it useful to filter based on RSVP status. "Give me only the calendar events I'm attending". But maybe that's a niche use case and instead that can be filtered on the client side since clients may want to show all RSVPs, regardless of status. What do you think?

Reply:
Ah yeah, that's probably a good use case for an index